include/zephyr/llext/llext.h

```diff
@@ -78,6 +86,30 @@ sys_slist_t *llext_list(void);
  */
 struct llext *llext_by_name(const char *name);
 
+/**
+ * @brief Iterate overall registered llext instances
+ *
+ * Calls a provided callback function for each registered extension or until the
+ * callback function returns a non-0 value.
+ *
+ * @param[in] fn callback function
+ * @param[in] arg a private argument to be provided to the callback function
+ * @retval 0 if no extensions are registered
+ * @retval value returned by the most recent callback invocation
+ */
+int llext_iterate(int (*fn)(struct llext *ext, void *arg), void *arg);
+
+/**
+ * @brief llext loader parameters
+ *
+ * These are parameters, not saved in the permanent llext context, needed only
+ * for the loader
+ */
+struct llext_load_param {
+	/** Should local relocation be performed */
+	bool relocate_local;
+};
```

Comment:
Most users will want to set this, so a non-zero value should be used as a default. In this case I suggest adding a `LLEXT_LOAD_PARAM_DEFAULT` macro so it's easy to initialize the struct correctly, and it's future proof when other special flags are added.

Reply:
@pillo79 how about I allow `NULL` and assume `true` by default? I do have to check for `NULL` anyway, it's missing now

Reply:
Nice addition! But I still think it would be helpful to have a future-proof public default (currently as small as `{ true }` :slightly_smiling_face: ), somewhat like the LLEXT_BUF_READER macro. 
If I were using NULL and later I wanted to customize some aspect of it, I could see defaults clearly documented. Otherwise I'd have to go and find out from the source what "NULL" actually meant.